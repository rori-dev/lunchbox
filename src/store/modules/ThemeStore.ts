import { ActionResult, store } from '@/store'
import { Action, Module, Mutation, VuexModule } from 'vuex-module-decorators'

// --- mdoel ---

export interface Theme {
  value: string
  text: string
}

export const THEME_RED = { value: 'theme-red', text: 'Red' }
export const THEME_GREEN = { value: 'theme-green', text: 'Green' }
export const THEME_BLUE = { value: 'theme-blue', text: 'Blue' }
export const THEME_BLUE_DARK = { value: 'theme-blue-dark', text: 'Blue (Dark)' }

// --- mutations ---

const SET_THEME = 'SET_THEME'

// --- module ---

@Module({ store, dynamic: true, namespaced: true, name: 'theme' })
export class ThemeStore extends VuexModule {
  // --- state ---

  theme: Theme = this.initialTheme()

  readonly themes: Theme[] = [THEME_RED, THEME_GREEN, THEME_BLUE, THEME_BLUE_DARK]

  @Mutation
  protected [SET_THEME](theme: Theme | string) {
    if (typeof theme === 'string')
      this.themes
        .filter(t => t.value === theme)
        .forEach(t => {
          this.theme = t
        })
    else this.theme = theme
  }

  @Action
  async updateTheme(theme: Theme | string): Promise<ActionResult> {
    try {
      this.context.commit(SET_THEME, theme)
      return ActionResult.success('Theme gespeichert')
    } catch (error) {
      return ActionResult.error('Theme speichern', error.message)
    }
  }

  protected initialTheme(): Theme {
    // if (window.matchMedia && window.matchMedia('(prefers-color-scheme: dark)').matches)
    //   return THEME_BLUE_DARK
    return THEME_BLUE
  }
}
