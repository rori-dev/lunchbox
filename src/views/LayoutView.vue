<template>
  <div class="layout-view">
    <Header />
    <Content />
  </div>
</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator'
import { Header, Content } from './layout/'

// vscode shows decorator errors -> https://github.com/vuejs/vetur/issues/815
@Component({
  components: {
    Header,
    Content,
  },
})
export default class LayoutView extends Vue {}
</script>
