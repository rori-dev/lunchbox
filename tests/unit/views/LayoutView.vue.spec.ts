import LayoutView from '@/views/LayoutView.vue'
import { createMock, mountUnit } from '@tests/unit/test-util'
import { ThemeStore } from '@/store/modules/ThemeStore'

describe('LayoutView', () => {
  test('renders snapshot', () => {
    const wrapper = mountUnit(LayoutView, {}, { provide })
    expect(wrapper.element).toMatchSnapshot()
  })
})

// --- mocks 'n' stuff

const mockStore = createMock(ThemeStore)
const provide = { themeStore: mockStore }
