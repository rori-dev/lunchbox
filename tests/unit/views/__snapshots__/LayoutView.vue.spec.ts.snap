// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LayoutView renders snapshot 1`] = `
<div
  class="layout-view h-full w-full font-sans antialiased"
>
  <header-stub />
   
  <content-stub />
</div>
`;
